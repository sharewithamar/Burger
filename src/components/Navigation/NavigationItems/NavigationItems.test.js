import React from 'react';
import { configure, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Adapter from 'enzyme-adapter-react-16';
import NavigationItems from './NavigationItems';
import NavigationItem from './NavigationItem/NavigationItem';
configure({ adapter: new Adapter() });
describe('<Navigation Items />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<NavigationItems />);
  });
  it('renders correctly enzyme', () => {
    const wrapper = shallow(<NavigationItems />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render two <NavigationItem/> if not authenticated', () => {
    // const wrapper = shallow(<NavigationItems />);
    expect(wrapper.find(NavigationItem)).toHaveLength(2);
  });
  it('should render three <NavigationItem/> if  authenticated', () => {
    //const wrapper = shallow(<NavigationItems isAuthenticated />);
    //  wrapper = wrapper = shallow(<NavigationItems isAuthenticated />);
    wrapper.setProps({ isAuthenticated: true });
    expect(wrapper.find(NavigationItem)).toHaveLength(3);
  });

  it('should contain Node', () => {
    wrapper.setProps({ isAuthenticated: true });
    expect(
      wrapper.contains(<NavigationItem link="/logout">Logout</NavigationItem>)
    ).toEqual(true);
  });
});
