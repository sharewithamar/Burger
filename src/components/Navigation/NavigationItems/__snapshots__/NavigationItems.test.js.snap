// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation Items /> renders correctly enzyme 1`] = `
<ul
  className="NavigationItems"
>
  <navigationItem
    exact={true}
    link="/"
  >
    Burger Builder
  </navigationItem>
  <navigationItem
    link="/auth"
  >
    Authenticate
  </navigationItem>
</ul>
`;
